### DISCUSSION TITLE
I don't understand how to override the size of input for the defined breakpoints.

### DISCUSSION THREAD

**Discussion by popoleeMaster** [mantinedev/mantine#4067] [Feature requests]
I want to make a design library using mantine as a base.

I would like to change the value of xs/sm/md/lg/xl of various components so I made the following.

```
const getSizes = (height: string) => ({
  height: height,
});

export const InputTheme: ThemeComponent = {
  sizes: {
    xs: () => ({
      root: {
        ...getSizes(rem(24)),
      },
    }),
    sm: () => ({
      root: {
        height: '54px',
      },
    }),
    md: () => ({
      root: {
        ...getSizes(rem(40)),
      },
    }),
    lg: () => ({
      root: {
        ...getSizes(rem(48)),
      },
    }),
    xl: () => ({
      root: {
        ...getSizes(rem(64)),
      },
    }),
  },
};
```

Then I use the above in 

```
export const DEFAULT_COMPONENTS: Record<string, ThemeComponent> = {
  Input: InputTheme
};


export const THEME: MantineThemeOverride = {
  components: DEFAULT_COMPONENTS,
};


```

and Finally 

```
    <MantineProvider withGlobalStyles withNormalizeCSS theme={THEME}>
      <Story />
    </MantineProvider>
```

But this do not override the size of the input, it still keep the one defined in mantine as default. I am pretty confused on the proper way to do

**Comment by rtivital** [comment]
You can change Input size by setting `size` prop. When you customize sizes on MantineProvider it does not assign any responsive values.

**Reply by popoleeMaster** [reply]
Thanks for the reply.

So there is no way to change the default value of `sm` `md` `lg` etc... so that if I do `<Input size='sm'/> it would use my value of `sm` assigned in the theme and not the default one of mantine ? 

Cause props is ok for implementing them in an app, but for a design library we want to provide our own theme and sizing for components to override the default one of mantine across the whole app

**Reply by rtivital** [reply]
You can add responsive values in your code as well as regular styles:

```tsx
export const InputTheme: ThemeComponent = {
  sizes: {
    xs: () => ({
      root: {
        ...getSizes(rem(24)),
       '@media (max-width: 50em)': { height: 100 }
      },
    }),
    //....
};
```
